layouts/product.njk

```diff
@@ -32,7 +32,11 @@
       {% endcall %}
     {% endblock %}
 
-    {{ xGovukRelatedNavigation(related) if related.sections[0].items | length > 0 }}
+    {#- Related items may contain many sections or none -#}
+    {%- set sections = related.sections or [related] -%}
+    {{ xGovukRelatedNavigation({
+      sections: sections
+    }) if related.sections or related.items | length > 0 }}
```

Comment:
I think the `xGovukRelatedNavigation` component should be able to handle being passed empty sections, no sections and lists with no items.

Can the logic be moved into the component template to simplify?

Reply:
Do any components in `govuk-frontend` validate params that are passed through? I don’t think they do (i.e. https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/breadcrumbs/template.njk)

Reply:
Ok, good shout. 

In govuk_publishing_components they do some checks:
https://github.com/alphagov/govuk_publishing_components/blob/main/app/views/govuk_publishing_components/components/_related_navigation.html.erb#L4

Reply:
A partial is probably sufficient.